stetho/src/main/java/com/facebook/stetho/inspector/network/ResponseBodyFileManager.java

```diff
@@ -56,19 +73,44 @@ public ResponseBodyData readFile(String requestId) throws IOException {
       }
       ResponseBodyData bodyData = new ResponseBodyData();
       bodyData.base64Encoded = firstByte != 0;
-      bodyData.data = readContentsAsUTF8(in);
+
+      if (mRequestIdMap.containsKey(requestId)) {
+        AsyncPrettyPrinter asyncPrettyPrinter = mRequestIdMap.get(requestId);
+        bodyData.data = prettyPrintContentWithTimeOut(asyncPrettyPrinter, in);
```

Comment:
Hmm.  The way this blocks, even if just for 10s, is problematic.  The caller that invokes this is, I believe, the stetho network thread caller.  This means that the entire data portion of the DevTools UI will be wedged for up to 10s, possibly multiple times over.  You can fix this by asynchronously processing this event on an executor.